pkg/cmd/pulumi/main.go

```diff
@@ -20,6 +20,8 @@ import (
 	"runtime"
 	"runtime/debug"
 
+	"github.com/pulumi/grpc-debug-interceptors"
```

Comment:
Can you make sure the grpc-debug-interceptors repo has reviews required and platform team ownership in team-management?

Reply:
I don't understand why this even needs to be a separate library? Can't it just be part of pkg, it's only the engine which needs to intercept and print this stuff.

Reply:
Yes if we adopt this PR I'll take care of that, otherwise can delete.

Reply:
Thanks for that comment, it might make it a little easier! It feels like the design is getting a little better/cleaner when GRPC options are surfaced as normal options in the utilities that sit under sdk/ and are populated at the call site with specific options. But it is a little tedious!

There's two places I can discern:

- rpcutil.Serve that starts gRPC servers
- plugin.Host that dials gRPC clients

I've refactored the rpcutil.Serve -> ServeWithOptions, but now looking at the second one a little bit. If I can't work it out quickly I can take your suggestion and go with some compromise.